src/pybind/mgr/dashboard/frontend/e2e/rgw/buckets.po.ts

```diff
@@ -1,5 +1,188 @@
 
-export class BucketsPage extends PageHelper {
-  pages = { index: '/#/rgw/bucket' };
+export class BucketsPageHelper extends PageHelper {
+  pages = {
+    index: '/#/rgw/bucket',
+    create: '/#/rgw/bucket/create'
+  };
+
+  create(name, owner) {
+    this.navigateTo('create');
+
+    // Enter in bucket name
+    element(by.id('bid')).sendKeys(name);
+
+    // Select bucket owner
+    element(by.id('owner')).click();
+    element(by.cssContainingText('select[name=owner] option', owner)).click();
+    expect(element(by.id('owner')).getAttribute('class')).toContain('ng-valid');
+
+    // Click the create button and wait for bucket to be made
+    const createButton = element(by.cssContainingText('button', 'Create Bucket'));
+    createButton.click().then(() => {
+      browser.wait(
+        Helper.EC.presenceOf(PageHelper.getTableCell(name)),
+        Helper.TIMEOUT,
+        'Timed out waiting for bucket creation'
+      );
+    });
+  }
+
+  edit(name, new_owner) {
+    this.navigateTo();
+
+    browser.wait(Helper.EC.elementToBeClickable(PageHelper.getTableCell(name)), 10000); // wait for table to load
+    PageHelper.getTableCell(name).click(); // click on the bucket you want to edit in the table
+    element(by.cssContainingText('button', 'Edit')).click(); // click button to move to edit page
+
+    expect(PageHelper.getBreadcrumbText()).toEqual('Edit');
+
+    const ownerDropDown = element(by.id('owner'));
+    ownerDropDown.click(); // click owner dropdown menu
+
+    element(by.cssContainingText('select[name=owner] option', new_owner)).click(); // select the new user
+    const editbutton = element(by.cssContainingText('button', 'Edit Bucket'));
+    editbutton.click().then(() => {
+      // wait to be back on buckets page with table visible
+      browser.wait(
+        Helper.EC.elementToBeClickable(PageHelper.getTableCell(name)),
+        10000,
+        'Could not return to buckets page and load table after editing bucket'
+      );
+
+      // click on edited bucket and check its details table for edited owner field
+      PageHelper.getTableCell(name).click();
+      const element_details_table = element
+        .all(by.css('.table.table-striped.table-bordered'))
+        .first();
+      expect(element_details_table.getText()).toMatch(new_owner);
+    });
+  }
+
+  delete(name) {
+    this.navigateTo();
+
+    // wait for table to load
+    browser.wait(Helper.EC.elementToBeClickable(PageHelper.getTableCell(name)), 10000);
+
+    PageHelper.getTableCell(name).click(); // click on the bucket you want to delete in the table
+    element(by.cssContainingText('.table-actions button.dropdown-toggle', '')).click(); // click toggle menu
+    element(by.cssContainingText('li.delete a', '')).click(); // click delete
+    // wait for pop-up to be visible (checks for title of pop-up)
+    browser
+      .wait(Helper.EC.visibilityOf(element(by.cssContainingText('.modal-title.pull-left', ''))), 5000)
```

Comment:
This changed in Bootstrap 4 
```suggestion
      .wait(Helper.EC.visibilityOf(element(by.cssContainingText('.modal-title.float-left', ''))), 5000)
```